branchLayout/api/src/main/java/com/virtuslab/branchlayout/api/BranchLayout.java

```diff
@@ -74,6 +74,19 @@ public boolean isEntryDuplicated(String branchName) {
     return null;
   }
 
+  public BranchLayout rename(String currentBranchName, String newBranchName) {
+    return new BranchLayout(rootEntries.flatMap(rootEntry -> rename(rootEntry, currentBranchName, newBranchName)));
+  }
+
+  private List<BranchLayoutEntry> rename(BranchLayoutEntry entry, String currentBranchName, String newBranchName) {
+    val newChildren = entry.getChildren().flatMap(child -> rename(child, currentBranchName, newBranchName));
+    if (entry.getName().equals(currentBranchName)) {
+      return List.of(new BranchLayoutEntry(newBranchName, entry.getCustomAnnotation(), entry.getChildren()));
```

Comment:
Nit:
```suggestion
      return List.of(entry.withName(newBranchName));
```
(add `@With` on `name`)